### PR TITLE
RHODS-2856: Opt in user data collection.

### PR DESCRIPTION
**Pull request by dlabaj**
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message - N/A
- [x] JIRA link(s):   https://issues.redhat.com/browse/RHODS-2856
- [x] The Jira story is acked
- [x] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)

This PR adds the ability for a RHODS admin to enable and disable user interaction data collection. 

Testing instructions:

RHODS admin testing:
1. Login to the RHODS dashboard as a RHODS admin.
2. Verify that Settings in the left hand navigation is present.
3. Verify that Cluster Settings is a menu item under Settings.
4. Navigate to Cluster Settings.
5. Verify that the "Allow collection of usage data" is checked or unchecked, based on redhat-ods-applications --> rhods-segment-key-config --> segmentKeyEnabled Config map in OpenShift.
6. Check or uncheck "Allow collection of usage data"
7. Verify that the config map in step 5 is updated.

Screen Shot 2022-02-21 at 9 12 43 AM

RHODS non admin testing:
1. Login to the RHODS dashboard as a non RHODS admin (RHODS developer).
2. Verify that Settings in the left hand navigation is NOT present.
3. Verify that Cluster Settings is NOT present.

### CODE REVIEW

**Review by DaoDaoNoCode — CHANGES_REQUESTED** [review]

File: backend/src/routes/api/segment-key/segmentKeyUtils.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export const getSegmentKey = async (fastify: KubeFastifyInstance): Promise<ODHSe
   try {
     const res = await coreV1Api.readNamespacedSecret('rhods-segment-key', namespace);
     const decodedSegmentKey = Buffer.from(res.body.data.segmentKey, 'base64').toString();
+    console.log(decodedSegmentKey);
```
[the remark this answers is not in the document]
Comment (dlabaj) [reply backend/src/routes/api/segment-key/segmentKeyUtils.ts]: Done

File: backend/src/routes/api/status/index.ts [hunk 2]
```diff
@@ -16,7 +17,27 @@ const status = async (
   if (!userName || userName === 'inClusterUser') {
     userName = 'kube:admin';
   }
-
+  const rbac = fastify.kube.rbac;
+  let isAdmin = false;
+  console.log('Running role bindings ...');
```
Comment (jeff-phillips-18) [inline backend/src/routes/api/status/index.ts:22]: remove console statement, or add dev check.

File: frontend/src/pages/clusterSettings/ClusterSettings.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.odh-cluster-settings {
+  .pf-c-form__group {
+    border: 1px solid var(--pf-global--BorderColor--100);
+    padding: 15px;
```
Comment (jeff-phillips-18) [inline frontend/src/pages/clusterSettings/ClusterSettings.scss:4]: PF var?

File: frontend/src/pages/clusterSettings/ClusterSettings.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    margin: 0px
+  }
+  .odh-number-input{
+    max-width: 40px;
+  }
+  .pf-c-input-group {
+    padding-top: 10px;
+    padding-bottom: 10px;
```
Comment (jeff-phillips-18) [inline frontend/src/pages/clusterSettings/ClusterSettings.scss:12]: PF Vars?

File: backend/src/routes/api/status/index.ts (hunk 2 above)
Comment (dlabaj) [reply backend/src/routes/api/status/index.ts:22]: Done.

**Review by jeff-phillips-18 — APPROVED** [review]

**Comment by jeff-phillips-18** [comment]
LGTM

File: frontend/src/pages/clusterSettings/ClusterSettings.scss (hunk 4 above)
Comment (dlabaj) [reply frontend/src/pages/clusterSettings/ClusterSettings.scss]: removed this it's not needed for this PR.

File: frontend/src/pages/clusterSettings/ClusterSettings.scss (hunk 3 above)
Comment (dlabaj) [reply frontend/src/pages/clusterSettings/ClusterSettings.scss]: Done

**Review by DaoDaoNoCode — APPROVED** [review]
LGTM

**Comment by jeff-phillips-18** [comment]
LGTM (again)

**Review by lucferbux — APPROVED** [review]
/lgtm

**Comment by lucferbux** [comment]
Added https://github.com/red-hat-data-services/odh-manifests/pull/204 for the rbac permission

File: backend/src/routes/api/status/index.ts [hunk 5]
```diff
@@ -16,7 +20,28 @@ const status = async (
...
+    const adminGroupResponse = await customObjectsApi.getClusterCustomObject(
+      'user.openshift.io',
+      'v1',
+      'groups',
+      adminGroup,
+    );
+    const adminUsers = (adminGroupResponse.body as groupObjResponse).users;
+    isAdmin = adminUsers.includes(userName);
+  } catch (e) {
+    console.log('Failed to get groups: ' + e.toString());
```
Comment (dlabaj) [inline backend/src/routes/api/status/index.ts:41]: @lucferbux Here's the fix you wanted.

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**